nifi-extension-bundles/nifi-network-bundle/nifi-network-processors/src/test/java/org/apache/nifi/processors/network/pcap/TestSplitPCAP.java

```diff
@@ -0,0 +1,159 @@
+            (long) 1713184965,
+            (long) 1000,
+            (long) 30,
+            (long) 30,
+            new byte[]{
+                0,  1,  2,  3,  4,  5,  6,  7,  8,  9,
+                10, 11, 12, 13, 14, 15, 16, 17, 18, 19,
+                20, 21, 22, 23, 24, 25, 26, 27, 28, 29,
+            }
+        );
+
+        this.invalidPacket = new Packet(
+            (long) 1713184965,
+            (long) 1000,
+            (long) 10,
+            (long) 10,
+            new byte[]{
+                0,  1,  2,  3,  4,  5,  6,  7,  8,  9,
+                10, 11, 12, 13, 14, 15, 16, 17, 18, 19,
+                20, 21, 22, 23, 24, 25, 26, 27, 28, 29,
+            }
+        );
+
+    }
+
+    @Test
+    public void testValidPackets() throws IOException {
+        TestRunner runner = TestRunners.newTestRunner(SplitPCAP.class);
+        runner.setProperty(SplitPCAP.PCAP_MAX_SIZE, "100");
+
+        List<Packet> packets = new ArrayList<>();
+        for (var loop = 0; loop < 3; loop++) {
+            packets.add(this.validPacket);
+        }
+
+        PCAP testPcap = new PCAP(this.hdr, packets);
+
+        runner.enqueue(testPcap.readBytesFull());
+
+        runner.run();
+
+        runner.assertTransferCount(SplitPCAP.REL_SPLIT, 3);
+        runner.assertTransferCount(SplitPCAP.REL_ORIGINAL, 1);
+        runner.assertQueueEmpty();
+    }
+
+    @Test
+    public void testInvalidPackets() throws IOException {
+        TestRunner runner = TestRunners.newTestRunner(SplitPCAP.class);
+        runner.setProperty(SplitPCAP.PCAP_MAX_SIZE, "50");
+
+        List<Packet> packets = new ArrayList<>();
+        for (var loop = 0; loop < 3; loop++) {
+            packets.add(this.invalidPacket);
+        }
+
+        PCAP testPcap = new PCAP(this.hdr, packets);
+
+        runner.enqueue(testPcap.readBytesFull());
+
+        runner.run();
+
+        runner.assertAllFlowFilesTransferred(SplitPCAP.REL_FAILURE, 1);
+        runner.assertQueueEmpty();
+    }
+
+    @Test
+    public void testPacketsTooBig() throws IOException {
+        TestRunner runner = TestRunners.newTestRunner(SplitPCAP.class);
+        runner.setProperty(SplitPCAP.PCAP_MAX_SIZE, "10");
+
+        List<Packet> packets = new ArrayList<>();
+        for (var loop = 0; loop < 3; loop++) {
+            packets.add(this.validPacket);
+        }
+
+        PCAP testPcap = new PCAP(this.hdr, packets);
+
+        runner.enqueue(testPcap.readBytesFull());
+
+        runner.run();
+
+        runner.assertAllFlowFilesTransferred(SplitPCAP.REL_FAILURE, 1);
+        runner.assertQueueEmpty();
+    }
+
+    @Test
+    public void testOneInvalidPacket() throws IOException {
+        TestRunner runner = TestRunners.newTestRunner(SplitPCAP.class);
+        runner.setProperty(SplitPCAP.PCAP_MAX_SIZE, "10");
+
+        List<Packet> packets = new ArrayList<>();
+        for (var loop = 0; loop < 3; loop++) {
+            packets.add(this.validPacket);
+        }
```

Comment:
Note the need to instantiate an `ArrayList` as `nCopies` returns an immutable `List`.
```suggestion
    @Test
    public void testOneInvalidPacket() {
        TestRunner runner = TestRunners.newTestRunner(SplitPCAP.class);
        runner.setProperty(SplitPCAP.PCAP_MAX_SIZE, "10");

        List<Packet> packets = new ArrayList<>(Collections.nCopies(3, this.validPacket));
```